### PR TITLE
Decompile func_80123B40

### PR DESCRIPTION
**Pull request by hohle**
Finishes off work originally started by @joshlory.

### CODE REVIEW

**Comment by hohle** [comment]
Thanks to @sozud for pointing out `memcpy`s could be simple assignments.

**Comment by Xeeynamo** [comment]
team work <3

File: src/dra/7E4BC.c [hunk 1]
```diff
@@ -1202,7 +1202,97 @@ void func_80123A60(Entity* entity) {
...
+
+        if (!(self->flags & 0x800000)) {
+            DestroyEntity(self);
+            return;
+        }
+
+        self->flags = 0x0C830000;
+        self->velocityY = FIX(-3.0);
+        SetSpeedX(-0x1AAAA);
+        self->ext.stub[0x28] = 0x80;
```
Comment (sozud) [inline src/dra/7E4BC.c:1233]: @Xeeynamo Do we need to create an ET_XXX here?

Comment (Xeeynamo) [reply src/dra/7E4BC.c:1233]: Yep 100%. The use of `self->ext.stub` should be disallowed and it's there just for padding

Comment (hohle) [reply src/dra/7E4BC.c:1233]: Sounds good. I'll see if any of the others are relevant otherwise create a new one.

Comment (Xeeynamo) [reply src/dra/7E4BC.c:1233]: We usually create new ones based on the context of the function. For example this one is calling `EntityPlayerBlinkWhite` with the same parameter. So maybe we can create a new struct called `ET_BlinkWhite blinkWhite` and use `self.ext.blinkWhite.unkA4`, or something like that.

**Comment by sozud** [comment]
Conflicts need to be resolved

**Pull request merged**